### DISCUSSION TITLE
Bootstrap 4 within Shadow-DOM/Web-Components

### DISCUSSION THREAD

**Discussion by rafalsk** [twbs/bootstrap#31612] [General]
We're having quite an issue with using Bootstrap within web-components. While the style-sheets import and render just fine within Shadow-DOM, the Boostrap scripts do not. We have tried multiple approaches to no avail. Including an attempt to simply import both the CSS and JavaScript file within the component. With Bootstrap 4 that wouldn't work. Shadow DOM is the only way for us to go since boundaries across components are one of our top priorities. ideas how to approach this?

**Comment by techboyg5** [comment]
For now, use the good old iframes.

**Comment by rafalsk** [comment]
That's not the way to go for us to reinvent the architecture and introduce cross-document messaging and serialization just to support a single UI-stack (from our perspective, for now, used for a couple of tabs and fancy drop-down menus).  Kindly thank you for your suggestion though.
